[WIP] Updated Demos for Polygon/Mainnet (#2897)
nucypher/nucypher: examples/finnegans_wake_demo/README.md

# Finnegan's Wake Demo

This illustrates Alice sharing data with Bob over the Threshold Network using proxy re-encryption (PRE),
without revealing private keys to intermediary entities.  For more detailed information see the [official documentation](https://docs.nucypher.com/en/latest/).

There are two version of the example, one using the decentralized network (ethereum/polygon), 
and a federated example using a local network.

### Decentralized Network Demo

First, configure the demo by making exporting environment variables
with your provider and wallet details.

```bash
export DEMO_L1_PROVIDER_URI=<YOUR ETH PROVIDER URL>
export DEMO_L2_PROVIDER_URI=<YOUR POLYGON PROVIDER URL>
export DEMO_L2_WALLET_FILEPATH=<YOUR WALLET FILEPATH>
export DEMO_ALICE_ADDRESS=<YOUR ALICE ETH ADDRESS>
```

Alternatively you can use the provided .env.template file by making a copy named .env
and adding your provider and wallet details.  To set the variables in your current session run:
`export $(cat .env | xargs)`

Optionally, you can change the network the demo is running on by changing the value of `L1_NETWORK` and `L2_NETWORK`.

> If you change these values be sure to also change `L1_PROVIDER_URI` and `L2_PROVIDER_URI` accordingly.

Reviewer: ```suggestion
If you change these values be sure to also change `DEMO_L1_PROVIDER_URI` and `DEMO_L2_PROVIDER_URI` accordingly.
```

Available options for `L1_NETWORK` are `ibex` or `mainnet`.

> Available options for `L2_NETWORK` are `mumbai` or `mainnet`

Reviewer: ```suggestion
Available options for `L2_NETWORK` are `mumbai` or `polygon`.
```

> Available options for `L1_NETWORK` are `ibex` or `mainnet`.
> Available options for `L2_NETWORK` are `mumbai` or `mainnet`

Reviewer: I wonder if something like this could be more clear by differentiating and grouping for testnet and mainnet:

```suggestion
For testnet: `L1_NETWORK` should be `ibex`, and `L2_NETWORK` should be `mumbai`.
For mainnet: `L1_NETWORK` should be `mainnet`, and `L2_NETWORK` should be `polygon`.
```

wdyt?

Reviewer: Also, do these env vars also need to be in the env template file?

Author: These ones are not env vars, just hardcoded as part of the example.

Reviewer: Filed #2919.
